static/opta.sh

```diff
@@ -0,0 +1,246 @@
+    DOCKERURL="https://download.docker.com/linux/static/stable/x86_64/docker-20.10.9.tgz"
+fi
+
+if  [ ! "$(command -v docker)" ]; then
+    echo "opta.sh requires Docker on your machine, please install docker."
+    exit 1
+fi
+
+##
+MYUID=`id -u ${USER}`
+MYGID=`id -g ${USER}`
+CONTAINERGID=$(( MYGID < 1000 ? 12345 : MYGID ))
+
+
+
+mkdir -p /tmp/opta
+
+cat << EOF > /tmp/opta/install.sh
+#! /usr/bin/env bash
+
+set -u
+
+NONINTERACTIVE=1
+
+echo "Going to install opta v$OPTAVERSION"
+
+echo $OPTAURL
+echo "Downloading installation package..."
+curl -s ${OPTAURL} -o /tmp/opta.zip --fail
+if [[ $? != 0 ]]; then
+  echo "Version $OPTAVERSION not found."
+  echo "Please check the available versions at https://github.com/run-x/opta/releases."
+  exit 1
+fi
+
+
+echo "Installing..."
+unzip -q /tmp/opta.zip -d /usr/local/bin/opta
+chmod u+x /usr/local/bin/opta/opta
+chown -R ${USER}:${USER} /usr/local/bin/opta
+
+EOF
+
+chmod +x /tmp/opta/install.sh
+
+cat << EOF > /tmp/opta/Dockerfile
+# base image
+FROM python:3.8.12-bullseye
+RUN apt-get update
+
+# aws cli
+RUN apt install unzip curl groff less -y
+RUN curl "$AWSCLI2URL" -o "awscliv2.zip"
+RUN unzip awscliv2.zip
+RUN ./aws/install
+RUN rm  awscliv2.zip
+
+# gcp cli 
+RUN curl "$GOOGLEURL" -o "googlecloudsdk.tar.gz"
+RUN tar -xvzf googlecloudsdk.tar.gz; sleep 1
+RUN rm googlecloudsdk.tar.gz
+ENV PATH="$PATH:/google-cloud-sdk/bin"
+
+# terraform
+RUN curl "$TERRAFORMURL" -o "terraform.zip"
+RUN unzip terraform.zip -d /usr/local/bin
+RUN chmod +x /usr/local/bin/terraform
+RUN rm terraform.zip
+
+# kubectl
+RUN curl -LO "$KUBECTLURL"
+RUN chmod +x kubectl
+RUN mv kubectl /usr/local/bin
+
+# helm - script is smart enough to handle different architectures
+RUN curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash
+
+# locales
+RUN apt-get install -y locales locales-all
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US.UTF-8
+ENV DOCKER_HOST=tcp://localhost:2375
+
+
+# build time env args
+
+ARG HOME
+ARG OPTAVERSION
+
+# add user
+RUN groupadd -g ${CONTAINERGID} ${USER}
+RUN useradd ${USER} -m -d ${HOME} -u ${MYUID} -g ${CONTAINERGID} 
+
+# docker (install after adding docker group above)
+RUN curl $DOCKERURL -o docker.tar.gz
+RUN tar xvzf docker.tar.gz
+RUN cp docker/* /usr/local/bin
+RUN rm docker.tar.gz
+RUN rm -rf docker/
+
+# clean up
+RUN apt-get clean
+
+# opta
+ADD install.sh /tmp/install.sh
+RUN /bin/bash -c "/tmp/install.sh"
+
+
+USER ${USER}
+WORKDIR ${HOME}
+
+ENTRYPOINT ["/usr/local/bin/opta/opta"]
+EOF
+
+echo "Building your Opta image"
+# Build the docker image
+docker build --quiet /tmp/opta \
+--build-arg HOME=${HOME} \
+--build-arg OPTAVERSION=${OPTAVERSION} \
+--tag=opta-${USER}:${OPTAVERSION} 
+echo "Done building Opta image"
+## Inject Opta-specific env vars from host env shell into the docker container
+declare -a opta_envvars=("AWS_ACCESS_KEY_ID" "AWS_SECRET_ACCESS_KEY" "MONGODB_ATLAS_PUBLIC_KEY" "MONGODB_ATLAS_PRIVATE_KEY")
```

Comment:
Could we inject all env vars instead? So we don't have to maintain this list?

Reply:
We could inject all of them. I listed them explicitly for the "principle of least privilege" but yea we could  inject all host env vars into the container.

Reply:
We need to add atleast `AWS_SESSION_TOKEN` to the list here. that fixed the issue I was having with this script.

Reply:
but adding all also seem fine. as there are gcp credentials and then aws has others like AWS_REGION and AWS_DEFAULT_REGION etc.

Reply:
All envs are now being exported into the container. Can you pull and try again @NitinAgg. Thanks